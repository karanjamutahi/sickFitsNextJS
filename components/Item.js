import React, { Component } from "react";
import PropTypes from 'prop-types';
import Link from 'next/link';

import Title from './styles/Title'; 
import ItemStyles from './styles/ItemStyles';
import PriceTag from './styles/PriceTag';
import formatMoney from '../lib/formatMoney';
import DeleteItem from '../components/DeleteItem'; 

class Item extends Component {
    static propTypes = {
        item: PropTypes.object.isRequired, 
    };

    render() {
        const { item } = this.props;
        return (
            <ItemStyles>
                {item.image && <img src={ this.props.format === 'large' ? item.largeImage :item.image} alt={item.title}/> }
                <Title>
                    <Link href="item/[pid]" as={`item/${item.id}`}>
                        <a>{item.title}</a>
                    </Link>
                </Title>
                <PriceTag>{formatMoney(item.price)}</PriceTag>
                <p>{item.description}</p>
                <div className="buttonList">
                    <Link href={{
                        //'update/[pid]' as={`update/${item.id}`}
                        pathname: 'update', 
                        query: { id: item.id },
                    }}><a>Edit</a></Link>
                    <Link href=''><a>Add To Cart</a></Link>
                    <DeleteItem id={item.id}>Delete Item</DeleteItem>
                </div>
            </ItemStyles>
        )
    }
}

export default Item;