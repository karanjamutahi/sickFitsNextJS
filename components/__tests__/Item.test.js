import Item from '../Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id:"sdlkajhsda0sa09dia",
    title: "Fake Item",
    description: "Just a Fake Item",
    image: "dog.jpg",
    largeImage: "large-dog.jpg",
    price:5999,
};

describe('Item Component', ()=>{
    const wrapper = shallow(<Item item={fakeItem}></Item>);

    it("Renders and matches the snapshot", () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    xit("renders and displays correctly", () =>{
        const priceTag = wrapper.find('PriceTag');
        expect(priceTag.children().text()).toBe("$59.99");
        expect(wrapper.find('Title a').children().text()).toBe(fakeItem.title);
    });

    xit("Renders the image properly", () => {
        const image = wrapper.find('img');
        expect(image.props().src).toBe(fakeItem.image);
        expect(image.props().alt).toBe(fakeItem.title);
    });

    xit("Renders out the buttons properly", () => {
        const buttonList = wrapper.find('.buttonList');
        expect(buttonList.children()).toHaveLength(3);
        expect(buttonList.find('Link').exists()).toBeTruthy();
        expect(buttonList.find('AddToCart').exists()).toBeTruthy();
        expect(buttonList.find('DeleteItem').exists()).toBeTruthy();
    });
});