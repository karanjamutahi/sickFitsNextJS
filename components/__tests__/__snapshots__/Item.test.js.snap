// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component Renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Fake Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      as="item/sdlkajhsda0sa09dia"
      href="item/[pid]"
    >
      <a>
        Fake Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $59.99
  </PriceTag>
  <p>
    Just a Fake Item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "sdlkajhsda0sa09dia",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="sdlkajhsda0sa09dia"
    />
    <DeleteItem
      id="sdlkajhsda0sa09dia"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
